pages/chaoss/visualizations/contrib_importance_over_time.py

```diff
@@ -0,0 +1,423 @@
+
+    # dynamically calculate the contributor importance over time for each of the action times and store results in df_final
+    (df_final['Commit'],
+    df_final['Issue Opened'], 
+    df_final['PR Opened'],
+    df_final['PR Comment'], 
+    df_final['PR Review']) = zip(*df_final.apply(lambda row: cntrb_importance_over_time(df, row.period_from, row.period_to, window_width, threshold), axis=1))
+    
+    return df_final
+
+def create_figure(df_final, step_size):
+    # create plotly express line graph
+    fig = go.Figure(
+        [
+            go.Scatter(name = 'Commit',
+                       x = df_final['period_from'],
+                       y = df_final['Commit'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[0])
+                       ), 
+            
+            go.Scatter(name = 'Issue Opened',
+                       x = df_final['period_from'],
+                       y = df_final['Issue Opened'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[1])
+                       ), 
+
+            go.Scatter(name = 'PR Opened',
+                       x = df_final['period_from'],
+                       y = df_final['PR Opened'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[2])
+                       ),
+            
+            go.Scatter(name = 'PR Request Comment',
+                       x = df_final['period_from'],
+                       y = df_final['PR Comment'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[3])
+                       ),
+
+            go.Scatter(name = 'PR Request Review',
+                       x = df_final['period_from'],
+                       y = df_final['PR Review'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[4])
+                       )
+                        
+        ], 
+    )
+    
+    # define x-axis and y-axis titles and intialize first x-axis tick to start at the user-inputted start_date
+    start_date = min(df_final['period_from'])
+
+    # update xaxes to display ticks
+    fig.update_xaxes(showgrid=True,
+                     ticklabelmode="period",
+                     tickangle=45,  
+                     dtick=f"M{step_size * 2}", 
+                     tickformat='%b %Y',)
+
+    
+    # hover template styling 
+    fig.update_traces(textposition="top right", 
+                      hovertemplate= "Date: %{x}" + "<br>Contributor Importance: %{y}<br><extra></extra>",
+                      )
+    
+    # layout syling
+    fig.update_layout(
+        xaxis_title=f'Timeline (stepsize = {step_size})',
```

Comment:
Right now this just renders as numerical:

"Timeline (stepsize =6)" so an annotation about the unit of step size would be nice.